src/bin/build_octree.rs

```diff
@@ -57,24 +57,26 @@ where
     match stream.size_hint() {
         Some(size) => println!(
             "Splitting {} which has {} points ({:.2}x MAX_POINTS_PER_NODE).",
-            node.id,
+            node_id,
             size,
             size as f64 / MAX_POINTS_PER_NODE as f64
         ),
         None => println!(
             "Splitting {} which has an unknown number of points.",
-            node.id
+            node_id
         ),
     };
 
+    let bounding_cube = node_id.bounding_cube(&build_meta.root_bounding_cube);
     stream.for_each(|p| {
-        let child_index = node.get_child_id_containing_point(&p.position);
+        let child_index = octree::ChildIndex::from_bounding_cube(&bounding_cube, &p.position);
         let array_index = child_index.as_u8() as usize;
         if children[array_index].is_none() {
             children[array_index] = Some(octree::NodeWriter::new(
```

Comment:
You pass in all members of BuildMeta. Maybe call it OctreeMeta and put it into the lib?